app/models/translation_string.rb

```diff
@@ -75,7 +76,13 @@ def update_localization
       raise("Localization for :#{tag.to_sym} doesn't exist!")
     end
 
-    data[tag.to_sym] = text
+    store_localization
+  end
+
+  # Update this string in the translations I18n is using.
+  def store_localization
+    I18n.backend.
+      store_translations(language.locale, { mo: { tag.to_sym => text } })
```

Comment:
The key here was storing the translation inside the `:mo` hash, not the whole translations hash! (there are non-MO translations in `I18n`)